src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -333,17 +402,24 @@ public FunctionPointerMethodSymbol SetNullabilityForReferenceTypes(Func<TypeWith
             CallingConvention callingConvention,
             RefKind refKind,
             TypeWithAnnotations returnTypeWithAnnotations,
+            ImmutableArray<CustomModifier> refCustomModifiers,
             ImmutableArray<TypeWithAnnotations> parameterTypes,
+            ImmutableArray<ImmutableArray<CustomModifier>> parameterRefCustomModifiers,
             ImmutableArray<RefKind> parameterRefKinds,
             CSharpCompilation compilation)
         {
             Debug.Assert(refKind != RefKind.Out);
-            RefCustomModifiers = getCustomModifierForRefKind(refKind, compilation);
+            Debug.Assert(parameterRefCustomModifiers.IsDefault || parameterRefCustomModifiers.Length == parameterTypes.Length);
+            RefCustomModifiers = refCustomModifiers.IsDefault ? getCustomModifierForRefKind(refKind, compilation) : refCustomModifiers;
```

Comment:
>refCustomModifiers [](start = 116, length = 18)

I assume this is the constructor to support public API for creating the symbol. Do you expect the user be responsible for adding In/Out modifiers when they want to specify calling convention modifiers?

Reply:
I'll leave a prototype comment. I'm currently unsure, I want to talk with Jason/Cyrus about that.

---
In reply to: [457729603](https://github.com/dotnet/roslyn/pull/46110#discussion_r457729603) [](ancestors = 457729603)